packages/react/src/reactrouterv6.tsx

```diff
@@ -0,0 +1,158 @@
+import { Transaction, TransactionContext } from '@sentry/types';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+import * as React from 'react';
+
+import { Action, Location } from './types';
+
+interface RouteObject {
+  caseSensitive?: boolean;
+  children?: RouteObject[];
+  element?: React.ReactNode;
+  index?: boolean;
+  path?: string;
+}
+
+type Params<Key extends string = string> = {
+  readonly [key in Key]: string | undefined;
+};
+
+interface RouteMatch<ParamKey extends string = string> {
+  params: Params<ParamKey>;
+  pathname: string;
+  route: RouteObject;
+}
+
+type UseLocation = () => Location;
+type UseNavigationType = () => Action;
+type CreateRoutesFromChildren = (children: JSX.Element[]) => RouteObject[];
+type MatchRoutes = (routes: RouteObject[], location: Location) => RouteMatch[];
+
+interface ReactRouterV6InstrumentationOptions {
+  customStartTransaction: (context: TransactionContext) => Transaction | undefined;
+  useLocation: UseLocation;
+  useNavigationType: UseNavigationType;
+  createRoutesFromChildren: CreateRoutesFromChildren;
+  matchRoutes: MatchRoutes;
+  startTransactionOnLocationChange?: boolean;
+  startTransactionOnPageLoad?: boolean;
+}
+
+const instrumentationOptions: Partial<ReactRouterV6InstrumentationOptions> = {
+  customStartTransaction: () => undefined,
+};
+
+export function reactRouterV6Instrumentation(
+  useLocation: UseLocation,
+  useNavigationType: UseNavigationType,
+  createRoutesFromChildren: CreateRoutesFromChildren,
+  matchRoutes: MatchRoutes,
+) {
+  return (
+    customStartTransaction: (context: TransactionContext) => Transaction | undefined,
+    startTransactionOnPageLoad = true,
+    startTransactionOnLocationChange = true,
+  ): void => {
+    instrumentationOptions.useLocation = useLocation;
+    instrumentationOptions.useNavigationType = useNavigationType;
+    instrumentationOptions.matchRoutes = matchRoutes;
+    instrumentationOptions.createRoutesFromChildren = createRoutesFromChildren;
+
+    instrumentationOptions.customStartTransaction = customStartTransaction;
+    instrumentationOptions.startTransactionOnLocationChange = startTransactionOnLocationChange;
+    instrumentationOptions.startTransactionOnPageLoad = startTransactionOnPageLoad;
```

Comment:
Let's lift these up into individual variables instead of being under the `instrumentationOptions` object. This way we save on bundle size because otherwise object keys like `customStartTransaction` or `startTransactionOnLocationChange` cannot be minified.

```suggestion
    useLocation = useLocation;
    useNavigationType = useNavigationType;
    matchRoutes = matchRoutes;
    createRoutesFromChildren = createRoutesFromChildren;

    customStartTransaction = customStartTransaction;
    startTransactionOnLocationChange = startTransactionOnLocationChange;
    startTransactionOnPageLoad = startTransactionOnPageLoad;
```